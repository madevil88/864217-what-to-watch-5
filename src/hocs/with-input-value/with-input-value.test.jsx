import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withInputValue from "./with-input-value";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withInputValue(MockComponent);

it(`withInputValue is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
