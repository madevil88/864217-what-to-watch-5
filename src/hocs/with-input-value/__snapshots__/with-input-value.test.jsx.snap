// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withInputValue is rendered correctly 1`] = `<div />`;
